src/python/WMCore/Storage/Backends/XRDCPImpl.py

```diff
@@ -169,19 +263,21 @@ def createDebuggingCommand(self, sourcePFN, targetPFN, options=None, checksums=N
         :targetPFN: str, destination PFN
         :options: str, additional options for copy command
         :checksums: dict, collect checksums according to the algorithms saved as keys
+        :authMethod: str, the authentication method to be preferentially used ("X509", "TOKEN", or None)
+        :forceMethod: bool, whether to force the use of the preferred authentication method, disabling the other 
         """
-        # Build the command for debugging purposes
-        copyCommandDict = self.buildCopyCommandDict(sourcePFN, targetPFN, options, checksums)
-        copyCommand = self.copyCommand.format_map(copyCommandDict)
+        # Construct xrdcp/xrdfs commands, but returning the debugging information instead of running it
+        return self.getFormattedCopyCommand(sourcePFN, targetPFN, options, checksums, authMethod, forceMethod, debug=True)
 
-        result = self.debuggingTemplate.format(copy_command=copyCommand, source=copyCommandDict['source'], destination=copyCommandDict['destination'])
-        return result
-
-    def removeFile(self, pfnToRemove):
+    def removeFile(self, pfnToRemove, authMethod=None, forceMethod=False):
         """
         _removeFile_
 
         """
+        authEnv = self.getAuthEnv(authMethod, forceMethod)
+        if authEnv:
+            self.xrdfsCmd = "%s %s" % (authEnv, self.xrdfsCmd)
+
         (_, host, path, _) = self.splitPFN(pfnToRemove)
         command = "%s %s rm %s" % (self.xrdfsCmd, host, path)
```

Comment:
Don't we have to account for the case where we are just debugging the command? Or the file removal is not part of the actual script we execute?

Reply:
AFAIK this `removeFile` method is not used as part of the `StageOutIml.__call__` that triggers the debugging mode.